code/game/objects/items/weapons/RCL.dm

```diff
@@ -0,0 +1,97 @@
+/obj/item/weapon/rcl
+	name = "rapid cable layer (RCL)"
+	desc = "A device used to rapidly deploy cables. It has a message engraved into the side, but it's chicken scratch."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "rcl-0"
+	item_state = "rcl-0"
+	opacity = 0
+	flags = FPRINT
+	siemens_coefficient = 1 //Not quite as conductive as working with cables themselves
+	force = 5.0 //Plastic is soft
+	throwforce = 5.0
+	throw_speed = 1
+	throw_range = 10
+	w_class = 3.0
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	origin_tech = "engineering=2;materials=4"
+	//var/active = 0 Depreciated; Leaving it because it is a useful framework tool if you want to make it automagically place on movement
+	var/obj/structure/cable/last = null
+	var/obj/item/stack/cable_coil/loaded = null
+
+/obj/item/weapon/rcl/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W,/obj/item/stack/cable_coil))
+		if(!loaded)
+			loaded = W
+			user.drop_item(W,src)
+		else
+			loaded.preattack(W,user,1)
+		update_icon()
+		user << "<span class='notice'>You add the cables to the [src]. It now contains [loaded.amount].</span>"
+	else
+		..()
+
+/obj/item/weapon/rcl/examine(mob/user)
+	..()
+	user << "<span class='info'>It contains [loaded.amount]/30 cables.</span>"
+
+/obj/item/weapon/rcl/update_icon()
+	if(!loaded)
+		icon_state = "rcl-0"
+		item_state = "rcl-0"
+		return
+	switch(loaded.amount)

```

Comment:
That doesn't really work perfectly for the reasons I said above. Take 0 case for example.

0 + (10 - 0 % 10) = 10, therefore it would never show as empty